File: src/main/python/com/nishant/sorting/selectionsort/selection_sort.py
import math

def selection_sort(arr):
    for start in range(len(arr)):
        min_num = math.inf
        min_index = -1

        for j in range(start, len(arr), 1):
            if arr[j] < min_num:
                min_num = arr[j]
                min_index = j

        arr[start], arr[min_index] = arr[min_index], arr[start]

if __name__ == "__main__":
    arr = [1, 2, 6, 4, 5]
    print(str(arr))
    selection_sort(arr)
    print(str(arr))

File: src/main/java/sorting/selectionsort/SelectionSort.java
package sorting.selectionsort;

import java.util.Arrays;

public class SelectionSort {

    public static void sort(int[] arr) {
        for (int start = 0; start < arr.length - 1; start++) {
            int min = Integer.MAX_VALUE, minIndex = -1;

            for (int i = start; i < arr.length; i++) {
                if (arr[i] < min) {
                    min = arr[i];
                    minIndex = i;
                }
            }

            swap(arr, start, minIndex);
        }
    }

    private static void swap(int[] arr, int pos1, int pos2) {
        int temp = arr[pos1];
        arr[pos1] = arr[pos2];
        arr[pos2] = temp;
    }

    public static void main(String[] args) {
        int[] arr = {1, 5, 3, 5, 3, 2, 4, 5, 2};
        System.out.println(Arrays.toString(arr));
        sort(arr);
        System.out.println(Arrays.toString(arr));
    }
}
